OpenRA.Mods.Common/Lint/CheckDeathTypes.cs

```diff
@@ -36,6 +36,10 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset ru
 				if (!targetable.Any())
 					continue;
 
+				var sa = actorInfo.Value.TraitInfos<SpawnActorOnDeathInfo>();
+				if (sa.Any())
+					deathTypes.AddRange(sa.Where(t => t.DeathType != null).Select(dt => dt.DeathType).ToList());
```

Comment:
This code won't run at all if the actor doesn't have any death type animations.  You will need to adjust the checks above to account for `SpawnActorOnDeath`.